kdc/kerberos5.c

```diff
@@ -60,15 +60,16 @@ realloc_method_data(METHOD_DATA *md)
     return 0;
 }
 
-static void
+static krb5_error_code
 set_salt_padata(METHOD_DATA *md, Salt *salt)
 {
-    if (salt) {
-       realloc_method_data(md);
-       md->val[md->len - 1].padata_type = salt->type;
-       der_copy_octet_string(&salt->salt,
-                             &md->val[md->len - 1].padata_value);
-    }
+    PA_DATA pa; /* do not free */
+
+    if (!salt)
+        return 0;
+    pa.padata_type = salt->type;
+    pa.padata_value = salt->salt;
+    return add_METHOD_DATA(md, &pa);
```

Comment:
which reminds me, `krb5_padata_add()` is a weird API, because it consumes its arguments...